lib/oauth.js

```diff
@@ -387,28 +393,26 @@ exports.Client = internals.Client = function (options) {
 
     this.provider = options.name;
     this.settings = {
-        temporary: internals.Client.baseUri(options.provider.temporary),
-        token: internals.Client.baseUri(options.provider.token),
         clientId: options.clientId,
         clientSecret: options.provider.signatureMethod === 'RSA-SHA1' ? options.clientSecret : internals.encode(options.clientSecret || '') + '&',
         signatureMethod: options.provider.signatureMethod
     };
 };
 
 
-internals.Client.prototype.temporary = function (oauth_callback) {
+internals.Client.prototype.temporary = function (uri, oauth_callback) {
 
     // Temporary Credentials (2.1)
 
     const oauth = {
         oauth_callback
     };
 
-    return this._request('post', this.settings.temporary, null, oauth, { desc: 'temporary credentials' });
+    return this._request('post', internals.Client.baseUri(uri), null, oauth, { desc: 'temporary credentials' });
 };
 
 
-internals.Client.prototype.token = function (oauthToken, oauthVerifier, tokenSecret) {
+internals.Client.prototype.token = function (uri, oauthToken, oauthVerifier, tokenSecret) {
```

Comment:
Same on every `Client` endpoint... I thought that was implicit... :-)

Reply:
I reverted the changes to these method signatures but kept their ability to resolve endpoints from a function, if a function is provided. It's a bit ugly since it potentially requires setting `client.settings.token` and `client.settings.temporary` in two passes, but I think it only applies to those who choose to upgrade to using a function instead of a string.

I didn't catch that `Client` was exported before (d'oh 😅). I think this should resolve the breaking change, right? Is there something else that I'm missing?